WordPress/openverse
# Update data refresh to use media tables

## Fixes



Fixes #2093 by @stacimc 
Fixes #2094 by @stacimc 

## Description



* Removes popularity calculation steps from the data refresh DAGs
* Updates the ingestion server to copy from the media tables, rather than the matviews

Most of the changes in the ingestion server add handling for the names of the upstream and downstream tables being different. Previously, we could always append `_view` to the end of the downstream table name to get the upstream table name. Now, we must do this for `audioset` but not for the media tables. (I.e.: the relevant tables in the API are `audio`, `image`, `audioset`. The corresponding catalog ones are `audio`, `image`, `audioset_view`).

When this is merged, data refreshes and popularity refreshes will officially be fully decoupled and we'll be able to drop the matviews.

This PR does not remove the matview locally or do other clean-up. This will come in a subsequent PR. The intention is to keep this PR as small as possible for review.


## Testing Instructions


Run `just recreate` from this branch. This will give you changes needed to the schema of the `audio` table and `audioset_view` and update your sample data. (These are prerequisite changes in `main`, which this branch has been rebased on).

Run a few provider DAGs to get some new data. I ran jamendo for a minute or two until I got 200 new audio records and several new audiosets, and SMK to get some images.

Run `just catalog/pgcli` and manually check the `COUNT` for each of the following:

```sql
SELECT COUNT FROM image;

SELECT COUNT FROM audio;

SELECT COUNT FROM audio WHERE audio_set_foreign_identifier IS NOT NULL;

SELECT COUNT FROM audioset_view;
```

Then run an image data refresh and audio data refresh locally and ensure they pass. Make sure you have the `create_filtered_image_index` and `create_filtered_audio_index` DAGs enabled before running the data refreshes.

Now confirm that the records are available in the API: run `just api/pgcli` and run the same `SELECT COUNT` queries you just ran in the catalog. The counts should all be identical.

**NOTE** This PR does not drop the materialized views. You can see that the records are being copied from the media tables by inspecting the logs in the ingestion server, but to be ABSOLUTELY certain you can run this in your catalog DB:

```psql
DROP MATERIALIZED VIEW audio_view;
DROP MATERIALIZED VIEW image_view;
```

And then re-run the data refreshes and ensure they still work.


## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
